contracts/PetPark.sol

```diff
@@ -1,7 +1,92 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.0;
-
+import "hardhat/console.sol";
 
 contract PetPark {
+    address public owner;
+
+    mapping(string => uint) public AnimalType;
+
+    mapping(uint => uint) public animalCounts;
+
+    mapping(address => User) public Users;
+
+	mapping(string => uint8) public Gender;
+
+    struct User {
+        bool hasEverBorrowed;
+        bool hasAnimal;
+        uint borrowedAnimalType;
+        uint gender;
+        uint age;
+    }
+
+    constructor() {
+        owner = msg.sender;
+
+        // AnimalType
+        AnimalType["Fish"] = 1;
+        AnimalType["Cat"] = 2;
+        AnimalType["Dog"] = 3;
+        AnimalType["Rabbit"] = 4;
+        AnimalType["Parrot"] = 5;
+
+        // Gender
+        Gender["Male"] = 0;
+        Gender["Female"] = 1;
+    }
+
+    modifier onlyOwner {
+        require(msg.sender == owner, "Not an owner");
+        _;
+   }
+
+    event Added(uint indexed animalType, uint indexed animalCount);
+
+    function add(uint _animalType, uint _count) public onlyOwner {
+        require(_animalType != 0, "Invalid animal");
+        animalCounts[_animalType] += _count;
+        emit Added(_animalType, animalCounts[_animalType]);
+    }
+
+    event Borrowed(uint indexed _animalType);
+
+    function borrow(uint _age, uint _gender, uint _animalType) public {
+        require(_age != 0, "Invalid Age");
+        require(_animalType != 0, "Invalid animal type");
+        require(animalCounts[_animalType] > 0, "Selected animal not available");
+        if(Users[msg.sender].age == 0) {
```

Comment:
Since `Users[msg.sender]` is being used multiple times it can be stored in a single memory variable. Later in the course you will learn that it is a gas optimisation technique

Reply:
Stored in `user memory sender = users[msg.sender];` within the `borrow` and `giveBackAnimal` functions!